### ISSUE TITLE
Query: Any attempt to submit upstream?

### ISSUE THREAD

**Issue by seanvk** [heechul/palloc#4]
Hello, I was curious as to whether there had been plans to submit this upstream?  I've not been able to track down any patch submissions for this project.

Thanks,

Sean

**Comment by heechul** [comment]
Thanks for your interests. I certainly considered making the patch upstream, but have not done that so for several reasons:

1. I was not able to reliably detect the bank mapping of recent Intel processors' memory controllers. Although the bank mapping detector itself is not part of PALLOC kernel patch, I figured that if we cannot apply the bank partitioning on ALL Intel processors, people may not be less willing to accept the patch.

2. When applied to partition the LLC, again, results from some recent Intel processors that we tested were a bit difficult to understand, most likely due to the undisclosed hash function based slice mapping.  

3. Perhaps most important reason is that I was (still am) concerned about the necessary amount of time spent to make it accepted. As a researcher in academia, it is a bit difficult to go through the time consuming kernel review process. 

In short, universal applicability and time are the main reasons. If you or anyone is interested in up-streaming the patch on his/her own, I will be more than happy to support that. 

Best
